api-bucket-lifecycle.go

```diff
@@ -48,15 +49,16 @@ func (c *Client) SetBucketLifecycle(ctx context.Context, bucketName string, conf
 	}
 
 	// Save the updated lifecycle.
-	return c.putBucketLifecycle(ctx, bucketName, buf)
+	return c.putBucketLifecycle(ctx, bucketName, buf, expiryRuleRemoved)
 }
 
 // Saves a new bucket lifecycle.
-func (c *Client) putBucketLifecycle(ctx context.Context, bucketName string, buf []byte) error {
+func (c *Client) putBucketLifecycle(ctx context.Context, bucketName string, buf []byte, expiryRuleRemoved bool) error {
 	// Get resources properly escaped and lined up before
 	// using them in http request.
 	urlValues := make(url.Values)
 	urlValues.Set("lifecycle", "")
+	urlValues.Set("expiryRuleRemoved", strconv.FormatBool(expiryRuleRemoved))
```

Comment:
```suggestion
	urlValues.Set("expiryRuleRemoved", strconv.FormatBool(expiryRuleRemoved))
```

We shouldn't add this by default no other S3 provider will recognize this and may even error out including AWS S3.

Reply:
If we move the flag under `lifecycle.Configuration` we dont need to set the explicitly anymore.